app/assets/javascripts/components/generic_object/generic-object-definition-component.js

```diff
@@ -76,6 +81,8 @@ function genericObjectDefinitionFormController(API, miqService, $q) {
 
     assignAllObjectsToKeyValueArrays(true);
 
+    vm.pictureReset = ! vm.pictureReset;
```

Comment:
clicking reset twice will remove the picture?

Reply:
Aaah, can you make this explicit please?

Because now it looks like you're just switching the var for no reason, and not actually using it.

(I mean, if you look for `pictureReset` in the code, you will just see that you set it to false, and that you negate it here. But never actually read that. .. So it's a bit hard to notice that you're using it for the side effect caused by changing it.)

I would make `$onChanges` do that only when `changes.pictureReset`, not on *any* change, unless you're relying on it happening for other changes.

Reply:
Addressed in fb22a12